hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/SparkHoodieTableFileIndex.scala

```diff
@@ -209,7 +215,8 @@ class SparkHoodieTableFileIndex(spark: SparkSession,
         s" after partition prune size is: ${prunedPartitionPaths.size}")
       prunedPartitionPaths
     } else {
-      partitionPaths
+      logInfo(s"No partition predicate provided, total partition size is: ${getAllQueryPartitionPaths.asScala.size}")
```

Comment:
Let's make sure we add comments elaborating on the lazy and non-lazy listing flows

Reply:
Yeah, I added some description on FileIndex constructor.